OneBlink.SDK/OneBlinkHttpClient.cs

```diff
@@ -64,5 +65,25 @@ public async Task<HttpContent> SendRequest(HttpRequestMessage httpRequestMessage
         return httpResponseMessage.Content;
       }
     }
+
+    public async Task<Stream> StreamRequest(HttpRequestMessage httpRequestMessage, string userToken = null)
+    {
+      // generate token
+      string token = !String.IsNullOrEmpty(userToken) ? userToken : Token.GenerateJSONWebToken(accessKey, secretKey, expiryInSeconds);
+      using (HttpClient httpClient = new HttpClient())
+      {
+        httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        httpClient.DefaultRequestHeaders.Add("User-Agent", ".Net OneBlink.SDK / " + GetType().Assembly.GetName().Version.ToString());
+
+        HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequestMessage);
+        if (!httpResponseMessage.IsSuccessStatusCode)
+        {
+          string result = await httpResponseMessage.Content.ReadAsStringAsync();
+          APIErrorResponse apiErrorResponse = JsonConvert.DeserializeObject<APIErrorResponse>(result);
+          throw new OneBlinkAPIException(apiErrorResponse);
+        }
+        return await httpResponseMessage.Content.ReadAsStreamAsync();
+      }

```

Comment:
Can this all be replaced by:

```c#
HttpContent httpContent = await SendRequest(httpRequestMessage, null);
return await httpContent.ReadAsStreamAsync()
```

I think there are some PDF functions that might be able to use this `StreamRequest()` as well.

Reply:
Yes, yes it can 🌁